R/user-access.R

```diff
@@ -0,0 +1,39 @@
+#' Get information about when users last accessed a dataset
+#'
+#' Useful to get dataset-user interaction metadata about all of the users that
+#' have access to a particular dataset.
+#'
+#' @param dataset_entity a dataset object or a dataset entity (e.g. a single
+#' entry from `selectDatasetCatalog()`)
+#' @param user_fields metadata fields about users to include in the dataframe
+#' (default: c("name", "email", "access_time"))
+#'
+#' @return a dataframe with information about all users for a particular dataset
+#' @export
+userInfo <- function (dataset_entity, user_fields = c("name", "email", "access_time")) {
+    users_catalog <- ShojiCatalog(crGET(paste0(self(dataset_entity),"users/")))
```

Comment:
Always avoid `paste0` to construct URLs in `crunch`. That's not how our API works. Use `shojiURL(dataset, "catalogs", "users")`.

Reply:
And I believe there already is a `UserCatalog` class, which may help here.